dags/als-kp-dataset-dag.py

```diff
@@ -271,32 +336,29 @@ def create_datasets(transformed_items: List[Dict[str, Any]], **context) -> str:
                 name=item["title"],
                 description=dataset_description,
                 parent_id=context["params"]["project_id"],
-            ).store(synapse_client=synapse_client)
+            )
+            dataset.annotations = {"source": "Critical Path Institute"}
+            dataset.store(synapse_client=synapse_client)
```

Comment:
nit: You shouldn't need to pass in `synapse_client` anywhere. In theory it should all "Just work" without it. See the call to `dataset_collection.store()`, where (I assume it is working) doesn't have it.